### PR TITLE
fix(api, shared-date): Deck extents fix to utilize padding for conflict checking in place of mount offsets

### PR DESCRIPTION
**Pull request by CaseyBatten**
# Overview
Cover RQA-3067

Update the deck extents checking logic to utilize padding at any given side of the robot to adhere to a series of limits on how far beyond the mechanical deck extents the pipette is allowed to travel.

In general, the padding system ensures the following:
- The front most nozzle must not cross the rear padding "line"
- The rear most nozzle must not cross the front padding "line"
- The left most nozzle must not cross the right padding "line"
- the right most nozzle must not cross the left padding "line"

In a practical example using deck slots padding system enforces the following ruleset:
Given the padding values accounted for against the deck extents, a 96ch pipette is within the Flex's extents when all of the following are true:
- Slot A1: Front right nozzle is within the rear and left-side padding limits
- Slot D1: Back right nozzle is within the front and left-side padding limits
- Slot A3 Front left nozzle is within the rear and right-side padding limits
- Slot D3: Back left nozzle is within the front and right-side padding limits
- Thermocycler Column A2: Front right nozzle is within padding limits

Padding measurements for the Flex utilized were identified by measuring a set of distance such that:
- The rear of the 96ch pipette did not cross wells in row F of a well plate in Deck slot row D
- The front of the 96ch did not cross wells in row G of a well plate in Deck slot row A
- The right side of the 96ch did not cross wells in Column 1 of a well plate in the Thermocycler slot
- The left side of the 96ch did not cross wells in Column 11 of a well plate in Deck slot column 3

Padding measurements for the OT-2 utilized were identified by measuring a set of distance such that:
- The rear of the 8ch did not cross wells in row G of a well plate in Deck slot row 1-2-3
- The front of the 8ch did not cross wells in row C of a well plate in Deck slot row 10-11-12

## Test Plan and Hands on Testing
The following cases must hold true in our units tests, analysis and execution:
- [x] well A1 of the thermocycler should be inaccessible to a 96ch in single-A12 partial tip pickup
- [x] well A2 of the thermocycler should be **accessible** to a 96ch in single-A12 partial tip pickup
- [x] well G1 of labware in slot A1 should be inaccessible to a 96ch in single-H12 partial config
- [x] well H1 of labware in slot A1 should be **accessible** to a 96ch in single-H12 partial config
- [x] well F1 of labware in slot D1 should be inaccessible to a 96ch in single-A12 partial config
- [x] well E1 of labware in slot D1 should be **accessible** to a 96ch in single-A12 partial config
- [x] well G10 of labware in slot A3 should be inaccessible to a 96ch in single-H1 partial config
- [x] well H11 of labware in slot A3 should be inaccessible to a 96ch in single-H1 partial config
- [x] well H10 of labware in slot A3 should be **accessible** to a 96ch in single-H1 partial config
- [x] well F10 of labware in slot D3 should be inaccessible to a 96ch in single-A1 partial config
- [x] well E11 of labware in slot D3 should be inaccessible to a 96ch in single-A1 partial config
- [x] well E10 of labware in slot D3 should be **accessible** to a 96ch in single-A1 partial config

## Changelog
The left and right mount offset data has been maintained for usefulness, but is now unused.
Padding offsets have been added to the the robot deck schema, and values have been added for Flex and OT2.
The engine now utilizes all four viable corners of a pipette to determine validity for deck actions (while in partial tip, this limitation for deck extents has been maintained for this PR).

## Review requests
Please let me know if this seems like an appropriate adjustment to how we are validating our extents logic. I believe this makes things rapidly communicable to developers, and simple to adjust in the future.

## Risk assessment
Medium/Low - This replaces the extents logic within the same version in which it was introduced, so we're mostly just effecting new behavior with this. This should enforce the same about of deck access we had before, but we should be sure to throughly test it.

### CODE REVIEW

**Review by sfoster1 — APPROVED** [review]
Approved with the same caveats - let's make sure the commit message has a lot of text about where these numbers came from and work on something where the numbers come from the drawings directly.

**Review by vegano1 — APPROVED** [review]
Changes make sense to me, last thing i would verify is just running these changes on an OT2.

File: shared-data/robot/schemas/1.json [hunk 1]
```diff
@@ -37,6 +37,29 @@
       "description": "The maximum addressable coordinates of the deck without instruments.",
       "$ref": "#/definitions/xyzArray"
     },
+    "paddingOffsets": {
```
Comment (sanni-t) [inline shared-data/robot/schemas/1.json:40]: I think calling this `paddingOffsetsFromDeckExtents` or something like that would help understand this property better.

File: shared-data/robot/schemas/1.json [hunk 2]
```diff
@@ -37,6 +37,29 @@
...
+      "description": "The distance from a given edge of a deck extent by which the maximum amount of travel outside the extents is limited.",
```
Comment (sanni-t) [inline shared-data/robot/schemas/1.json]:
```suggestion
      "description": "The distance from a given edge of a deck extent by which the maximum amount of travel is limited.",
```

File: shared-data/robot/schemas/1.json [hunk 3]
```diff
@@ -37,6 +37,29 @@
...
+      "type": "object",
+      "required": ["rear", "front", "leftSide", "rightSide"],
+      "properties": {
+        "rear": {
+          "description": "The padding distance from the rear of the extents which the front nozzles must not exceed.",
+          "type": "number"
```
Comment (sanni-t) [inline shared-data/robot/schemas/1.json]:
```suggestion
          "description": "The padding distance from the rear edge of the deck extents which the front nozzles of a pipette must not exceed.",
          "type": "number"
```
maybe?

File: shared-data/robot/schemas/1.json [hunk 4]
```diff
@@ -37,6 +37,29 @@
...
+        "front": {
+          "description": "The padding distance from the front of the extents which the rear nozzles must not exceed.",
+          "type": "number"
+        },
+        "leftSide": {
+          "description": "The padding distance from the left of the extents which the right-most nozzles must not exceed.",
+          "type": "number"
+        },
+        "rightSide": {
+          "description": "The padding distance from the right of the extents which the left-most nozzles must not exceed.",
```
Comment (sanni-t) [inline shared-data/robot/schemas/1.json]: similar wording as above

**Review by sanni-t — APPROVED** [review]
Thanks for the changes and thorough testing. LGTM!
Definitely should move on to using true machine limits.

**Pull request merged**